pkg/server/cache/entrycache/fullcache_ds.go

```diff
@@ -0,0 +1,189 @@
+package entrycache
+
+import (
+	"context"
+	"time"
+
+	"github.com/golang/protobuf/ptypes/timestamp"
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/spiffe/spire/pkg/server/api"
+	"github.com/spiffe/spire/pkg/server/plugin/datastore"
+	"github.com/spiffe/spire/proto/spire/common"
+	"github.com/spiffe/spire/proto/spire/types"
+)
+
+const (
+	agentPageSize = 5000
+	entryPageSize = 5000
```

Comment:
Hmm, wonder if we should make these an order of magnitude smaller. SPIFFE IDs can technically be 2048 bytes a piece, which means if there were large registration entries, we could end up surpassing the 4MiB limit.

Reply:
Good point. If we make the assumption that registration entries are generally expected to be smaller than 8 KB, then I think a page size of 500 would be workable.

That seems like a reasonable tradeoff between supporting large registration entries and supporting many registration entries, which would result in more paged queries with a lower page size.

Reply:
This reminds me... https://github.com/spiffe/spire/issues/1446 :)